### ISSUE TITLE
Social attention: Modeling attention in human crowds

### ISSUE THREAD

**Issue by hsanchez** [hsanchez/paper-memory#7]
## TL;DR

Social Attention is a novel trajectory prediction model that captures the relative importance of each person when navigating in the crowd, irrespective of their proximity.

### Paper Link

- [arxiv.org/abs/1710.04689](https://arxiv.org/abs/1710.04689)
- [Social Attention web](https://paperswithcode.com/paper/social-attention-modeling-attention-in-human)
- [github.com/huang-xx/STGAT](https://github.com/huang-xx/STGAT)

### Author/Institution

- Anirudh Vemula (Carnegie Mellon University)
- Katharina Muelling (Carnegie Mellon University)
- Jean Oh (Carnegie Mellon University)

## Overview

This work proposes an approach that captures the relative importance of each person when navigating in the crowd, irrespective of their proximity. This approach uses a feedforward, fully differentiable, and jointly trained recurrent neural network (RNN) mixture to model trajectories of all humans in the crowd, addressing both spatial and temporal aspects of the problem. The human-human interactions are modeled using a _soft attention model_ over all humans in the crowd, thereby not restricting the approach with the local neighborhood assumption. Experiments show that the model captures the influence of each person on the other, the nature of their interaction, and predicts their future trajectories.  Social Attention is capable of predicting human trajectories more accurately than the state-of-the-art approach on two publicly available real-world crowd datasets.

The paper seeks to answer the question: Which surrounding agents do humans attend to while navigating a crowd? It is hypothesized that the representation of trajectories learned by Social Attention enables helps reason about the importance of surrounding agents better than only considering spatially local agents. This approach builds on Structural-rnn (Vemula et al. 2018) for this purpose.

## Contributions and Distinctions from Previous Works

- Focus on the problem of human trajectory prediction in crowded spaces.
- Addresses both spatial and temporal aspects of the human-human trajectory prediction problem by using a feedforward, fully differentiable, and jointly trained recurrent neural network (RNN) mixture.
- Conduct an empirical evaluation of the model to demonstrate the effectiveness of Social Attention at predicting human trajectories, clearly outperforming the state-of-the-art approach on two publicly available datasets.
- 

### Related areas

Related areas include modeling human interactions for navigation, human trajectory prediction, and Spatio-temporal models.

- Modeling human interactions for navigation.  Social forces can model can capture simple interactions but can’t model complex crowd behavior such as cooperation. Interacting Gaussian Processes can model the joint distribution of trajectories of all interacting agents in the crowd. However, these approaches model interactions based on relative distances and orientations, ignoring other features such as velocity and acceleration.
- Human trajectory prediction. Motion patterns of pedestrians in videos are learned using Gaussian Processes and observed trajectories are clustered to capture patterns. These motion patterns capture navigation behavior such as static obstacle avoidance, but they ignore human-human interactions. Recent work considers all agents in the environment, rather than just the local neighborhood, using attention. However, the attention used is hard-wired based on proximity rather than learned from data.
- Spatio-temporal models. This work formulates the task of human trajectory prediction using spatio-temporal graphs. Traditionally, graphical models such as Conditional Random Fields are used to model problems using this Spatio-temporal formulation, e.g., Robotics. These models used variations of Structural RNN(S-RNN). In this work, they use a variant of S-RNN.

## Methods

In this work, the aim is to model the influence of all agents in the crowd by learning an attention model over the agents. In other words, this work seeks to answer the question: Which surrounding agents do humans attend to, while navigating a crowd? It is hypothesized that the representation of trajectories learned by the attention model enables effective reasoning of the importance of surrounding agents better than only considering spatially local agents.

Specifically, it uses a feedforward, fully differentiable, and jointly trained RNN mixture that predicts both their future locations and captures human-human interactions. The adopted architecture differs from the S-RNN architecture, by introducing an attention module to compute a "soft attention" over hidden states of neighboring spatial edge RNNs for each node. Social LSTM is used as the baseline to compare the performance of this architecture and method against the state of the art.

## Results

The proposed Social attention model performs better than Social LSTM consistently across all the crowd sets in both the metrics. By learning the relative importance of each pedestrian in the crowd from data, Social Attention results in more accurate predictions. Ultimately, there is evidence that the proposed method outperforms the state-of-the-art approach in prediction errors, on two publicly available datasets.
